### PR TITLE
RemoteNode Ping

### PR DESCRIPTION
**Pull request by shargon**
Needed for https://github.com/neo-project/neo-cli/issues/287

### CODE REVIEW

**Comment by igormcoelho** [comment]
Do you think we can use akka to test it? Last time I was struggling to make all mailboxes to communicate, but I failed... looks like ActorSystem Config (string mailbox configs) need to be passed somehow to TestKit Sys (which is the ActorSystem used by tests). If we manage somehow to do that, everything will be much easier to test on dev time.

File: neo/Network/P2P/RemoteNode.cs [hunk 1]
```diff
@@ -146,10 +147,20 @@ protected override void OnReceive(object message)
...
+            var ping = new PingPayload();
+            lastPingNonce = ping.Nonce;
+
+            SendMessage(Message.Create(MessageCommand.Ping, ping));
+        }
+
         private void OnPingPayload(PingPayload payload)
         {
-            if (payload.LastBlockIndex > LastBlockIndex)
+            if (payload.Nonce == lastPingNonce && payload.LastBlockIndex > LastBlockIndex)
```
Comment (erikzhang) [inline neo/Network/P2P/RemoteNode.cs]: Why checking the `lastPingNonce`?

File: neo/Network/P2P/RemoteNode.cs [hunk 2]
```diff
@@ -146,10 +147,20 @@ protected override void OnReceive(object message)
             }
         }
 
+        public void Ping()
+        {
+            var ping = new PingPayload();
```
Comment (erikzhang) [inline neo/Network/P2P/RemoteNode.cs]: Should use `PingPayload.Create()`.

File: neo/Network/P2P/RemoteNode.cs (hunk 1 above)
Comment (shargon) [reply neo/Network/P2P/RemoteNode.cs]: What is the use of the nonce?

Comment (erikzhang) [reply neo/Network/P2P/RemoteNode.cs]: `Nonce` is used to identify which ping the corresponding pong corresponds to.

**Comment by erikzhang** [comment]
How do you use this in cli?

**Comment by shargon** [comment]
Fetching remote nodes and calling `Ping` on `show state`

**Comment by erikzhang** [comment]
The easiest way to do that is send messages to `LocalNode` directly. It will broadcast the message to all connected nodes.

https://github.com/neo-project/neo/blob/d4111532434e8a75c75e82303d7b16a5965850bf/neo/Network/P2P/LocalNode.cs#L142-L144

**Comment by shargon** [comment]
Ok, i will change the way!

**Pull request closed without merging**